src/TitleFactory.php

```diff
@@ -2,12 +2,23 @@
 
 namespace Hotmeteor\Titles;
 
+use Hotmeteor\Titles\Enums\Style;
+use Exception;
+
 class TitleFactory
 {
     public function __construct(public string $lang)
     {
     }
 
+    public static function title(string $value, Style $style): string
+    {
+        return match($style) {
+            Style::APA => (new TitleFactory)->apa($value);
+            default => throw new Exception("The style \"{$style->name}\" is currently not implemented");
+        };
```

Comment:
Alternatively, instead of an `enum`, a style could have its own class, implementing a `WritingStyle` interface, which can be extended to be used as `CustomWritingStyle`